erpnext/accounts/doctype/shipping_rule/shipping_rule.py

```diff
@@ -113,7 +113,31 @@ def add_shipping_rule_to_tax_table(self, doc, shipping_amount):
 		else:
 			shipping_charge["tax_amount"] = shipping_amount
 			shipping_charge["description"] = self.label
-			doc.append("taxes", shipping_charge)
+
+			# Check if already tax is presnt & add new shipping taxes
+			if doc.get("taxes"):
+				for i, tax in enumerate(doc.get("taxes")):
+					tax = tax.as_dict()
```

Comment:
- Are we assuming that only the previously added tax would be relevant and set it aftre clearing the taxes table ?

Reply:
Yes

Reply:
Why ?